File: dotnet-elasticemail-examples/Examples/EmailVerification.cs
namespace ElasticEmailExamples;

using ElasticEmail.Api;
using ElasticEmail.Client;

public static class EmailVerification
{
    // Usage: dotnet run -- email-verification someone@example.com
    public static async Task RunAsync(string[] args)
    {
        var verificationsApi = new VerificationsApi(Ee.Config());
        var email = args.Length > 0 ? args[0] : Ee.To;

        // Email verification is a paid feature. Accounts without it get a 4xx here.
        try
        {
            await verificationsApi.VerificationsByEmailPostAsync(email);
            var data = await verificationsApi.VerificationsByEmailGetAsync(email);

            Console.WriteLine("=== Verification result ===");
            Console.WriteLine($"Email:       {data.Email}");
            Console.WriteLine($"Result:      {data.Result}");
            Console.WriteLine($"Reason:      {data.Reason}");
            Console.WriteLine($"Disposable:  {data.Disposable}");
            Console.WriteLine($"Role:        {data.Role}");
            if (!string.IsNullOrEmpty(data.SuggestedSpelling))
            {
                Console.WriteLine($"Did you mean: {data.SuggestedSpelling}");
            }
        }
        catch (ApiException e)
        {
            Ee.Fail("verify email", e);
        }
    }
}

File: elixir-elasticemail-examples/examples/email_verification.exs
alias ElasticEmail.Env

# Usage: mix run examples/email_verification.exs someone@example.com
email = List.first(System.argv()) || Env.to()
email_path = ElasticEmail.path_segment(email)

# Email verification is a paid feature. Accounts without it get a 4xx here.
with {:ok, _} <- ElasticEmail.post("/verifications/#{email_path}", %{}),
     {:ok, v} <- ElasticEmail.get("/verifications/#{email_path}") do
  IO.puts("=== Verification result ===")
  IO.puts("Email:       #{v["Email"]}")
  IO.puts("Result:      #{v["Result"]}")
  IO.puts("Reason:      #{v["Reason"] || ""}")
  IO.puts("Disposable:  #{v["Disposable"]}")
  IO.puts("Role:        #{v["Role"]}")
  if v["SuggestedSpelling"], do: IO.puts("Did you mean: #{v["SuggestedSpelling"]}")
else
  error ->
    IO.puts(:stderr, "Error verifying email: #{ElasticEmail.format_error(error)}")
    System.halt(1)
end
